src/cljam/io/csi.clj

```diff
@@ -0,0 +1,72 @@
+(ns cljam.io.csi
+  "Reader of a CSI format file."
+  (:require [cljam.io.util.bgzf :as bgzf]
+            [cljam.io.util.lsb :as lsb]
+            [cljam.io.util.bin :as util-bin])
+  (:import java.util.Arrays
+           [java.io DataInputStream IOException]
+           [cljam.io.util.chunk Chunk]))
+
+(deftype CSI [n-ref min-shift depth bidx loffset]
+  util-bin/IBinningIndex
+  (get-chunks [_ ref-idx bins]
+    (into [] (mapcat (get bidx ref-idx) bins)))
+  (get-min-offset [_ ref-idx beg bins]
+    (let [oct-cumulative-sums (->> (range (inc depth))
+                                   (map #(bit-shift-left 1 (* % 3)))
+                                   (reductions +))
+          min-index (->> (take-while #(<= % (last bins)) oct-cumulative-sums)
+                         last)]
+      (or (->> bins
+               (drop-while #(>  min-index %))
```

Comment:
Extra spaces between `>` and `min-index`. And `partial` is preferable. https://guide.clojure.style/#partial
```suggestion
               (drop-while (partial > min-index))
```